mirrord/layer/tests/common/mod.rs

```diff
@@ -1383,20 +1383,44 @@ pub fn get_env(
             MIRRORD_LAYER_INTPROXY_ADDR.to_string(),
             intproxy_addr.to_string(),
         ),
+        (
+            LayerConfig::RESOLVED_CONFIG_ENV.to_string(),
+            config.encode().unwrap(),
+        ),
+        #[cfg(target_os = "macos")]
         (
             "DYLD_INSERT_LIBRARIES".to_string(),
             dylib_path.to_str().unwrap().to_string(),
         ),
+        #[cfg(target_os = "linux")]
         (
             "LD_PRELOAD".to_string(),
             dylib_path.to_str().unwrap().to_string(),
         ),
+        #[cfg(all(target_os = "macos", target_arch = "aarch64"))]
         (
-            LayerConfig::RESOLVED_CONFIG_ENV.to_string(),
-            config.encode().unwrap(),
+            "MIRRORD_MACOS_ARM64_LIBRARY".to_string(),
+            arm64_dylib_path().to_str().unwrap().to_string(),
         ),
```

Comment:
I think we should document somewhere why we need this to run the tests (especially as it is not needed for CI until we get arm64 darwin runners, so could be easily removed by mistake). I think here could be a place to document it. The contributing guide might also make sense, up to you which one to choose.

Reply:
Cool, I will add some note here